mock-kubernetes-client/client/client.go

```diff
@@ -0,0 +1,668 @@
+/*
```

Comment:
The benefit of writing our own mock instead of using the official one (that we use in the plugin tests) is that it's more flexible?

Reply:
Is there an official mock for the controller-runtime client?

Reply:
I've taken a look at that mock, and there are a handful of things that our tests are relying on that the official mock doesn't implement: generations, UIDs, field selectors in list calls, service IPs. There are some further things that we could add to a custom mock to improve our testing, such as pod IPs, scheduling failures, and termination failures. I think it's worth investing in our own mock.

Reply:
Seems reasonable.